modules/openapi-generator/src/test/resources/3_0/issue_5381.yaml

```diff
@@ -0,0 +1,134 @@
+openapi: 3.0.1
+info:
+  title: ByRefOrValue
+  description: >
+    This tests for a oneOf interface representation
+  version: 0.0.1
+servers:
+  - url: "http://localhost:8080"
+tags:
+  - name: Foo
+paths:
+  /foo:
+    get:
+      tags:
+        - Foo
+      summary: GET all Foos
+      operationId: getAllFoos
+      responses:
+        '200':
+          $ref: '#/components/responses/200FooArray'
+    post:
+      tags:
+        - Foo
+      summary: Create a Foo
+      operationId: createFoo
+      requestBody:
+        $ref: '#/components/requestBodies/Foo'
+      responses:
+        '201':
+          $ref: '#/components/responses/201Foo'
+
+components:
+  schemas:
+    Entity:
+      type: object
+      allOf:
+        - "$ref": "#/components/schemas/Addressable"
+        - "$ref": "#/components/schemas/Extensible"
+
+    EntityRef:
+      description: Entity reference schema to be use for all entityRef class.
+      type: object
+      properties:
+        name:
+          type: string
+          description: Name of the related entity.
+        '@referredType':
+          type: string
+          description: The actual type of the target instance when needed for disambiguation.
+      allOf:
+        - $ref: '#/components/schemas/Addressable'
+        - "$ref": "#/components/schemas/Extensible"
+
+
+    Addressable:
+      type: object
+      properties:
+        href:
+          type: string
+          description: Hyperlink reference
+        id:
+          type: string
+          description: unique identifier
+      description: Base schema for adressable entities
+    Extensible:
+      type: object
+      properties:
+        "@schemaLocation":
+          type: string
+          description: A URI to a JSON-Schema file that defines additional attributes
+            and relationships
+        "@baseType":
+          type: string
+          description: When sub-classing, this defines the super-class
+        "@type":
+          type: string
+          description: When sub-classing, this defines the sub-class Extensible name
+      required:
+        - '@type'
+
+    FooRefOrValue:
```

Comment:
Given you have a mustache which attempts to determine if it's an interface or class depending on how many nested levels, thoughts on having another container? There is nesting for `allOf` here but not `oneOf`

```
OuterFooRefOrValue:
  type: object
  oneOf:
    - $ref: "#/components/schemas/FooRefOrValue"
    - $ref: "#/components/schemas/AnotherRefOrValue"
  discriminator:
    propertyName: "@type"
      mapping:
        FooRefOrValue: "#/components/schemas/FooRefOrValue"
        AnotherRefOrValue: "#/components/schemas/AnotherRefOrValue"
```